app/test/vesting_test.go

```diff
@@ -0,0 +1,542 @@
+package app_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	tmtime "github.com/tendermint/tendermint/types/time"
+	"google.golang.org/grpc"
+)
+
+func TestVestingModule(t *testing.T) {
+	suite.Run(t, new(VestingModuleTestSuite))
+}
+
+type VestingModuleTestSuite struct {
+	suite.Suite
+
+	// Regular accounts
+	accounts       []string
+	mut            sync.Mutex
+	accountCounter int
+}
+
+func (s *VestingModuleTestSuite) SetupSuite() {
+	t := s.T()
+	t.Log("setting up vesting module test suite")
+	s.accounts = testfactory.GenerateAccounts(300)
+}
+
+func (s *VestingModuleTestSuite) unusedAccount() string {
+	s.mut.Lock()
+	acc := s.accounts[s.accountCounter]
+	s.accountCounter++
+	s.mut.Unlock()
+	return acc
+}
+
+func (s *VestingModuleTestSuite) NewNetworkWithGenesisOpt(kr keyring.Keyring, genesisOpts ...testnode.GenesisOption) (cctx testnode.Context, rpcAddr, grpcAddr string) {
+	tmCfg := testnode.DefaultTendermintConfig()
+	tmCfg.Consensus.TargetHeightDuration = time.Millisecond * 1
+	appConf := testnode.DefaultAppConfig()
+
+	for _, name := range s.accounts {
+		testfactory.NewBaseAccount(kr, name)
+	}
+
+	cctx, rpcAddr, grpcAddr = testnode.NewNetwork(s.T(), testnode.DefaultParams(), tmCfg, appConf, []string{}, genesisOpts...)
```

Comment:
[blocking]
this has gotten updated recently, apologies for the refactor. this should probably look like this now
https://github.com/celestiaorg/celestia-app/blob/3f8c8a84ab2767dd5b0f5c22c7aab6653a788047/app/test/square_size_test.go#L46-L49

Reply:
does this portion of the refactor make sense @mojtaba-esk ? it won't look exactly like the above but similar. Essentially we can fit everything that need to change into that config

let me know if this doesn't make sense, happy to walk through it

Reply:
yes it does make the code simpler. I merged it into a copy of this branch and will open another PR shortly